src/xengine/netio/ioclient.cpp

```diff
@@ -153,6 +159,14 @@ void CSocketClient::AsyncConnect(const tcp::endpoint& epRemote, CallBackConn fnC
 {
     sockClient.async_connect(epRemote, boost::bind(&CSocketClient::HandleConnCompleted, this,
                                                    fnConnected, boost::asio::placeholders::error));
+    int keepAlive = 1;
+    int keepIdle = 60 * 4;
+    int keepInterval = 5;
+    int keepCount = 2;
+    setsockopt(sockClient.native_handle(), SOL_SOCKET, SO_KEEPALIVE, (void*)&keepAlive, sizeof(keepAlive));
+    setsockopt(sockClient.native_handle(), SOL_TCP, TCP_KEEPIDLE, (void*)&keepIdle, sizeof(keepIdle));
```

Comment:
You could Remove Socket KeepAlive on linux, CI report error on MacOSX. There is no reason to continue using TCP KeepAlive after you added ping pong message.